src/plang/ExpressionNode.h

```diff
@@ -1,109 +1,24 @@
-#ifndef plang_Expression_h
-#define plang_Expression_h
+#ifndef plang_ExpressionNode_h
+#define plang_ExpressionNode_h
 
 #include <plang/def.h>
 #include <plang/Token.h>
+#include <plang/Operand.h>
+#include <plang/Operator.h>
 
 namespace plang {
-
-////////////////////////////////////////////////////////////////////////////////
-/// \class Operator
-////////////////////////////////////////////////////////////////////////////////
-class Operator
-{
-	friend class ExpressionNode;
-
-public:
-	enum Type
+	class ExpressionNode
 	{
-		kTypeInvalid = 0,
-		kTypeAdd,
-		kTypeSubtract,
-		kTypeCount
+		public:
+			enum Type
+			{
+				kTypeOperand = 0,
+				kTypeOperator
+			};
+
```

Comment:
@john-chapman Ideally I'd like to see this enummy Type comparison stuff disappear, in favour of simple type comparison.  Couldn't figure out how to get that working, though!  All depends on whether or not we want to follow a strictly OO paradigm with the actual dev of the system..?